### PR TITLE
Initialize the API setup for CostManagement CRD and the necessary role & role bindings.

### PR DESCRIPTION
**Pull request by chambridge**
* Add spec configuration values; assuming all values are optional
* Create role and role bindings for CRD and openshift-config data

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+*NOTE* If you have enabled webhooks in your deployments, you will need to have cert-manager already installed
+in the cluster or `make deploy` will fail when creating the cert-manager resources.
+
+Verify that the korekuta-operator-go is up and running:
+
+```console
+$ oc get deployment
+```
+
+## Create a Memcached CR
```
Comment (abaiken) [inline README.md:137]:
```suggestion
## Create a CostManagement CR
```

File: README.md [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+
+Next, install the Operator SDK CLI using the following [documentation](https://sdk.operatorframework.io/docs/installation/install-operator-sdk/). The operator is currently being built with the v0.19 release of the operator-sdk.
+
+To build the manager binary you can execute the following make command:
+
+```
+make manager
+```
+
+To build the docker image you can execute the follwoing make command:
```
Comment (abaiken) [inline README.md:33]:
```suggestion
To build the docker image you can execute the following make command:
```

File: README.md [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+
+## Create a CostManagement CR
+
+Create the CR:
+
+```sh
+$ oc apply -f config/samples/cost-mgmt_v1alpha1_costmanagement.yaml
+```
+
+Ensure that the korekuta operator creates the deployment for the sample CR with the correct size:
```
Comment (abaiken) [inline README.md:145]: I don't think that this applies to us (I think it is for the Memcached Operator).

Comment (chambridge) [reply README.md:145]: you are right. I'll remove. Clearly didn't catch all the memcache references.

Comment (chambridge) [reply README.md]: done

**Comment by maskarb** [comment]
Running `make docker-build` blew away the `roles.yaml`. I think if you add the following to `controllers.costmanagement_controllers.go` (line-37), the `roles.yaml` will autogenerate correctly:

```
// +kubebuilder:rbac:groups=cost-mgmt.openshift.io,resources=costmanagements,verbs=get;list;watch;create;update;patch;delete
// +kubebuilder:rbac:groups=cost-mgmt.openshift.io,resources=costmanagements/status,verbs=get;update;patch
// +kubebuilder:rbac:groups=config.openshift.io,resources=proxies;networks,verbs=get;list
// +kubebuilder:rbac:groups=authorization.k8s.io,resources=subjectaccessreviews;tokenreviews,verbs=create
// +kubebuilder:rbac:groups=core,resources=namespaces,verbs=get;list
// +kubebuilder:rbac:groups=core,namespace=openshift-config,resources=secrets,verbs=get
// +kubebuilder:rbac:groups=core,namespace=openshift-cost,resources=pods;services;services/finalizers;endpoints;persistentvolumeclaims;events;configmaps;secrets,verbs=create;delete;get;list;patch;update;watch
```

**Comment by chambridge** [comment]
Added the lines code changes mentioned @maskarb. Nice catch 👍

**Comment by chambridge** [comment]
With the latest I can deploy:
```
make deploy IMG="quay.io/project-koku/korekuta-operator-go:api-and-role-init"               2 ↵  419  12:01:33
go: creating new go.mod: module tmp
go: found sigs.k8s.io/controller-tools/cmd/controller-gen in sigs.k8s.io/controller-tools v0.3.0
/Users/chambrid/go/bin/controller-gen "crd:trivialVersions=true" rbac:roleName=manager-role webhook paths="./..." output:crd:artifacts:config=config/crd/bases
cd config/manager && /usr/local/bin/kustomize edit set image controller=quay.io/project-koku/korekuta-operator-go:api-and-role-init
/usr/local/bin/kustomize build config/default | kubectl apply -f -
namespace/system unchanged
customresourcedefinition.apiextensions.k8s.io/costmanagements.cost-mgmt.openshift.io configured
role.rbac.authorization.k8s.io/korekuta-operator-go-leader-election-role unchanged
role.rbac.authorization.k8s.io/korekuta-operator-go-manager-role configured
clusterrole.rbac.authorization.k8s.io/korekuta-operator-go-manager-role configured
clusterrole.rbac.authorization.k8s.io/korekuta-operator-go-proxy-role unchanged
clusterrole.rbac.authorization.k8s.io/korekuta-operator-go-metrics-reader unchanged
rolebinding.rbac.authorization.k8s.io/korekuta-operator-go-leader-election-rolebinding unchanged
rolebinding.rbac.authorization.k8s.io/korekuta-operator-go-manager-rolebinding unchanged
clusterrolebinding.rbac.authorization.k8s.io/korekuta-operator-go-manager-rolebinding unchanged
clusterrolebinding.rbac.authorization.k8s.io/korekuta-operator-go-proxy-rolebinding unchanged
service/korekuta-operator-go-controller-manager-metrics-service unchanged
deployment.apps/korekuta-operator-go-controller-manager unchanged
/usr/local/bin/kustomize build config/openshift-config | kubectl apply -f -
role.rbac.authorization.k8s.io/korekuta-operator-go-manager-config-role created
rolebinding.rbac.authorization.k8s.io/korekuta-operator-go-manager-config-rolebinding created
```

And you will see the following logs in the deployed operator:
```
I0915 16:01:30.153784       1 request.go:621] Throttling request took 1.066262878s, request: GET:https://172.25.0.1:443/apis/autoscaling.openshift.io/v1?timeout=32s
2020-09-15T16:01:31.381Z	INFO	controller-runtime.metrics	metrics server is starting to listen	{"addr": "127.0.0.1:8080"}
2020-09-15T16:01:31.381Z	INFO	setup	starting manager
I0915 16:01:31.382572       1 leaderelection.go:242] attempting to acquire leader lease  openshift-cost/91c624a5.openshift.io...
2020-09-15T16:01:31.382Z	INFO	controller-runtime.manager	starting metrics server	{"path": "/metrics"}
I0915 16:01:49.002753       1 leaderelection.go:252] successfully acquired lease openshift-cost/91c624a5.openshift.io
2020-09-15T16:01:49.004Z	DEBUG	controller-runtime.manager.events	Normal	{"object": {"kind":"ConfigMap","namespace":"openshift-cost","name":"91c624a5.openshift.io","uid":"b4950751-ab8b-4589-ae77-99cc0564b0e5","apiVersion":"v1","resourceVersion":"493745"}, "reason": "LeaderElection", "message": "korekuta-operator-go-controller-manager-c88b94c66-rh5mn_0cb8cb77-67cd-4f38-a4f6-e64a51a272a2 became leader"}
2020-09-15T16:01:49.003Z	INFO	controller-runtime.controller	Starting EventSource	{"controller": "costmanagement", "source": "kind source: /, Kind="}
2020-09-15T16:01:49.187Z	INFO	controller-runtime.controller	Starting Controller	{"controller": "costmanagement"}
2020-09-15T16:01:49.187Z	INFO	controller-runtime.controller	Starting workers	{"controller": "costmanagement", "worker count": 1}
2020-09-15T16:10:29.049Z	DEBUG	controller-runtime.controller	Successfully Reconciled	{"controller": "costmanagement", "request": "openshift-cost/example"}
```

File: README.md [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+repository sites such quay.io or hub.docker.com. This example uses quay.
+
+Build the image:
+```sh
+$ export USERNAME=<quay-username>
+
+$ make docker-build IMG=quay.io/$USERNAME/korekuta-operator-go:v0.0.1
+```
+
+Push the image to a repository:
```
Comment (abaiken) [inline README.md:103]:
```suggestion
Push the image to a repository and make sure to set the repository to public:
```

File: config/samples/cost-mgmt_v1alpha1_costmanagement.yaml [hunk 5]
```diff
@@ -3,5 +3,3 @@ kind: CostManagement
 metadata:
   name: costmanagement-sample
 spec:
```
Comment (abaiken) [inline config/samples/cost-mgmt_v1alpha1_costmanagement.yaml:5]: whenever I deploy this weith the empty spec, I get this error: 
```
$ oc apply -f config/samples/cost-mgmt_v1alpha1_costmanagement.yaml
The CostManagement "costmanagement-sample" is invalid: spec: Invalid value: "null": spec in body must be of type object: "null"
```
this is resolved by removing the spec field

Comment (chambridge) [reply config/samples/cost-mgmt_v1alpha1_costmanagement.yaml]: removed the `spec:` line.

**Review by abaiken — APPROVED** [review]
LGTM! Might want to let @maskarb finish testing to see if he finds any gotchas!

**Review by abaiken — COMMENTED** [review]
Also might just be me but my `config/manager/kustomization.yaml` still shows up as modified even though its in the .gitignore.

**Comment by chambridge** [comment]
@abaiken yeah confused about the .gitignore, must have done something stupid 🤷

**Pull request merged**